Storage/SQL/BrowserDB.swift

```diff
@@ -430,12 +430,20 @@ extension BrowserDB: Queryable {
 
     func queryReturnsResults(sql: String, args: Args?=nil) -> Deferred<Maybe<Bool>> {
         return self.runQuery(sql, args: args, factory: { row in true })
-         >>== { deferMaybe($0[0] ?? false) }
+            >>== { if ($0.status == .Success) && ($0.count > 0) {
+                return deferMaybe($0[0] ?? false)
```

Comment:
This continues to result in the error `Cannot convert call result type 'Deferred<Maybe<Cursor<_>>>' to expected type 'Deferred<Maybe<_>>'`

I'm not sure why this is happening, possibly `$0[0]` can be accessed only when the status is a `.Success`. But, how does the compiler correlate these?

So, how about we go with: 
```
            >>== { guard $0.status == .Success  else { return deferMaybe(false) }
                return deferMaybe($0[0] ?? false)
        }

```